### ISSUE TITLE
File size limit on uploads?

### ISSUE THREAD

**Issue by cognizen** [vote539/socketio-file-upload#3]
Again, great stuff.  This npm has proven to be incredibly useful to this point!

I am running into an issue that seems to be related to the socket timing out when I upload lager files - say, more than 25MB.  I know I know - huge for the internet but I'm combining this with fluent-ffmpeg to convert and edit video files so in some cases they can get pretty large.

Any idea on how to force the socket to not time out?  Is this a socket.io or file-upload thing - if socket, is there a way to over-ride it to stay active?

**Comment by MaffooBristol** [comment]
I'm going to have a pop at chunking uploads. Will see if I have any success. Wondering if having a Flash fallback for browsers with dubious FileReader support may be a good idea, in the same way that SoundManager 2 polyfills browsers that don't support HTML5 audio with Flash bindings... will report back, but it's a major roadblock in my application and reliance on SIOFU! Thanks

**Comment by MaffooBristol** [comment]
Ps, @vote539 - I get an error that says `Uncaught InvalidStateError: Failed to execute 'readAsArrayBuffer' on 'FileReader': The object is already busy reading Blobs.` every time I upload a file. Do you reckon it may be related to the `readAsArrayBuffer` issues that have been mentioned before?

**Comment by MaffooBristol** [comment]
Okay, I've got chunking to work. It's very slightly similar to what the guy on that SO post was saying, combined with loads of other bits I've scoured from the internet, but is relatively simple. I think using the progress events aren't really going to cut it as there's very little control to be had over them. Also I think that the chunks have to be multiples of 8 or something for Uint8Array() to work, and what the browser returns as its progress event chunks are pretty random, which is what's causing the TypeError. So instead it does the chunking manually and then generates a new FileReader object for each given chunk, and uses its `onload` event to recursively process the next chunk (calling `siofu_progress` along the way) until it hits the end and fires the `siofu_done` callback.

Anyway, enough waffling. I'm going to clean up my mess-tacular code, test it and then do a PR, and hopefully we can get this thing sorted once and for all! :grinning: :tada:
